docs/BEMAN_PROCESS.md

```diff
@@ -0,0 +1,49 @@
+<!--
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+# The Beman Process
+
+This document specifies stages and best practices for Beman Project libraries development.
+Its goal is to create consistency facilitating the evaluation of, and contribution to Beman libraries.
+
+## Introduction
+
+### Changing this document
+
+This is a living document that adapts to evolving best practices and community
+needs. To make changes:
+
+1. Create a [discourse topic](https://discourse.bemanproject.org/latest) detailing the changes
+   and how it aligns with the core principles.
+2. After some community discussion, create a PR with the actual changes on
+   [GitHub](https://github.com/bemanproject/beman) with the `!!! Beman leads required !!!`
+   label. The PR should also link to the discourse topic.
+3. Continue discussions on the PR and discourse topic.
+4. Await a leads a decision based on the community feedback.
```

Comment:
Can we replace leads / leader with maintainer everywhere in the README? wdyt?

Reply:
No. Not all Beman members are leaders or codeowners.

Only few of the them are leaders (e.g., David, Jeff, Inbal).